WebTests/IAuthor_TestClass.cs

```diff
@@ -0,0 +1,191 @@
+﻿// <copyright file="IAuthor_TestClass.cs" company="My company">
+//     Copyright (c) My company". All rights reserved.
+// </copyright>
+
+namespace WebTests
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using WebApi.BookService;
+    using WebLib;
+    using WebLib.Models;
+
+    /// <summary>
+    /// IAuthor test class
+    /// </summary>
+    [TestClass]
+    public class IAuthor_TestClass
+    {
+        /// <summary>
+        /// Data provider object
+        /// </summary>
+        private static readonly IDataProvider DataProvider; // stylecop thinks, that it must be with upper case
+
+        /// <summary>
+        /// IAuthor object
+        /// </summary>
+        private IAuthor authorsObject;
+
+        /// <summary>
+        /// Initializes static members of the <see cref="IAuthor_TestClass"/> class
+        /// </summary>
+        static IAuthor_TestClass()
+        {
+            DataProvider = new DataProviderList();
+        }
+
+        /// <summary>
+        /// Test initialize method
+        /// </summary>
+        [TestInitialize]
+        public void TestClassInitialize()
+        {
+            this.authorsObject = new LibraryList(DataProvider);
+        }
+
+        /// <summary>
+        /// Add author test
+        /// </summary>
+        [TestMethod]
+        public void AddAuthor()
+        {
+            // Arrange
+            Author expected = new Author("Hello", "Hello", "Hello");
+
+            // Act
```

Comment:
here is a mistake in test logic. Your test is saying that it is testing AddAuthor. So you should call only the AddAuthor. But not addAuthor and GetAuthor. 

I will write an example in the chat how it should be tested